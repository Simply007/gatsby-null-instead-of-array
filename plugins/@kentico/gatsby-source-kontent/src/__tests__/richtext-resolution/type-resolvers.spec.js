const { KontentTestHttpService }
  = require('@kentico/kontent-test-http-service-js');
const { ContentItem, TypeResolver } = require('@kentico/kontent-delivery');

// Project ID 71be1cca-0be3-0159-fdfe-6cf11b092e68
const richtextFakeItemsResponse =
  require('./richtextFakeItemsResponse.json');
const richtextFakeTypesResponse =
  require('./richtextFakeTypesResponse.json');

const { sourceNodes } = require('../../gatsby-node');


describe(`Rich text resolution reference in modular content`, () => {
  const fakeRichTextResponseConfig = new Map();
  fakeRichTextResponseConfig.set(
    /https:\/\/deliver.kontent.ai\/.*\/items/,
    {
      fakeResponseJson: richtextFakeItemsResponse,
      throwError: false,
    });

  fakeRichTextResponseConfig.set(
    /https:\/\/deliver.kontent.ai\/.*\/types/,
    {
      fakeResponseJson: richtextFakeTypesResponse,
      throwError: false,
    });

  const dummyCreateNodeID = jest.fn();
  dummyCreateNodeID.mockImplementation((input) => `dummy-${input}`);

  const createNodeMock = jest.fn();
  const createTypesMock = jest.fn();
  const mockedSchema = { buildObjectType: jest.fn((input) => ({
    data: input,
  }))};

  const actions = {
    actions: {
      createNode: createNodeMock,
      createTypes: createTypesMock,
    },
    createNodeId: dummyCreateNodeID,
    schema: mockedSchema,
  };

  class LandingPageImageSection extends ContentItem {
    constructor() {
      super({
        richTextResolver: (_contentItem, _context) =>
          '###landing_page_image_section###',
      });
    }
  }

  class Project extends ContentItem {
    constructor() {
      super({
        urlSlugResolver: (_link, _context) => ({ url: '###projectlink###' }),
        richTextResolver: (_contentItem, _context) =>
          '###project###',
      });
    }
  }

  const deliveryClientConfig = {
    projectId: 'dummyProject',
    typeResolvers: [
      new TypeResolver('landing_page_image_section', (rawData) =>
        new LandingPageImageSection(rawData)
      ),
      new TypeResolver('project', (rawData) =>
        new Project(rawData)),
    ],
    httpService: new KontentTestHttpService(
      fakeRichTextResponseConfig
    ),
  };

  const pluginConfiguration = {
    deliveryClientConfig,
    languageCodenames: ['default'],
  };

  it('resolves values using resolvers in rich text', async () => {
    await sourceNodes(actions, pluginConfiguration);

    const calls = createNodeMock.mock.calls;
    expect(calls).toMatchSnapshot();
  });
});
