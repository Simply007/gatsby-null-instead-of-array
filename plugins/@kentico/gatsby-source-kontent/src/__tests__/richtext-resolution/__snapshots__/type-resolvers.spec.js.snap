// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rich text resolution reference in modular content resolves values using resolvers in rich text 1`] = `
Array [
  Array [
    Object {
      "children": Array [],
      "contentItems___NODE": Array [
        "dummy-kentico-kontent-item-simple-landing-page-default",
      ],
      "elements": Array [
        Object {
          "codename": "content",
          "name": "Content",
          "options": Array [],
          "type": "rich_text",
        },
      ],
      "id": "dummy-kentico-kontent-type-landing-page",
      "internal": Object {
        "contentDigest": "a0ac33b7e607597366ab8ba9d6c24889",
        "type": "KontentTypeLandingPage",
      },
      "parent": null,
      "system": Object {
        "codename": "landing_page",
        "id": "b300f06d-d1e7-40b4-9dfc-e065e0455593",
        "lastModified": "2019-10-09T10:20:42.6215148Z",
        "name": "Landing page",
      },
      "usedByContentItems___NODE": Array [],
    },
  ],
  Array [
    Object {
      "children": Array [],
      "contentItems___NODE": Array [
        "dummy-kentico-kontent-item-n26ddfb46-c2de-010e-3ad9-ec9aa8955144-default",
      ],
      "elements": Array [
        Object {
          "codename": "title",
          "name": "Title",
          "options": Array [],
          "type": "text",
        },
        Object {
          "codename": "images",
          "name": "Images",
          "options": Array [],
          "type": "asset",
        },
      ],
      "id": "dummy-kentico-kontent-type-landing-page-image-section",
      "internal": Object {
        "contentDigest": "b8c43d50ff8a1c089e6ceae4a2944567",
        "type": "KontentTypeLandingPageImageSection",
      },
      "parent": null,
      "system": Object {
        "codename": "landing_page_image_section",
        "id": "45e6379c-858c-49aa-b293-b1218581931b",
        "lastModified": "2018-12-31T11:37:53.7435133Z",
        "name": "Landing page image section",
      },
      "usedByContentItems___NODE": Array [],
    },
  ],
  Array [
    Object {
      "children": Array [],
      "contentItems___NODE": Array [
        "dummy-kentico-kontent-item-main-project-default",
      ],
      "elements": Array [
        Object {
          "codename": "content",
          "name": "Content",
          "options": Array [],
          "type": "text",
        },
      ],
      "id": "dummy-kentico-kontent-type-project",
      "internal": Object {
        "contentDigest": "8524000f6fc39895b271183bcddbd9d2",
        "type": "KontentTypeProject",
      },
      "parent": null,
      "system": Object {
        "codename": "project",
        "id": "ec706130-b0dd-401c-ae1a-10f2eeb2eb89",
        "lastModified": "2019-10-09T10:20:12.7155113Z",
        "name": "Project",
      },
      "usedByContentItems___NODE": Array [],
    },
  ],
  Array [
    Object {
      "children": Array [],
      "contentType___NODE": "dummy-kentico-kontent-type-project",
      "elements": Object {
        "content": Object {
          "name": "Content",
          "type": "text",
          "value": "Main project content",
        },
      },
      "id": "dummy-kentico-kontent-item-main-project-default",
      "internal": Object {
        "contentDigest": "76d61969e0b504ce90f234ee5c246223",
        "type": "KontentItemProject",
      },
      "otherLanguages___NODE": Array [],
      "parent": null,
      "preferred_language": "default",
      "system": Object {
        "codename": "main_project",
        "id": "1a3aa6bd-7b4b-486f-86ab-7bfe359ad614",
        "language": "default",
        "lastModified": "2018-12-28T12:53:59.075Z",
        "name": "Main project",
        "sitemapLocations": Array [],
        "type": "project",
      },
      "usedByContentItems___NODE": Array [
        "dummy-kentico-kontent-item-simple-landing-page-default",
      ],
    },
  ],
  Array [
    Object {
      "children": Array [],
      "contentType___NODE": "dummy-kentico-kontent-type-landing-page",
      "elements": Object {
        "content": Object {
          "images": Array [
            Object {
              "description": null,
              "height": 1700,
              "imageId": "98bf37b8-8be3-45e9-b284-293e205bf9a3",
              "url": "https://assets-eu-01.kc-usercontent.com:443/71be1cca-0be3-0159-fdfe-6cf11b092e68/6c7c8367-f38c-41a1-a41d-91a71dcc073a/Back%203_4%20right.jpg",
              "width": 3010,
            },
          ],
          "linkedItemCodenames": Array [
            "main_project",
            "n26ddfb46_c2de_010e_3ad9_ec9aa8955144",
          ],
          "linked_items___NODE": Array [
            "dummy-kentico-kontent-item-main-project-default",
            "dummy-kentico-kontent-item-n26ddfb46-c2de-010e-3ad9-ec9aa8955144-default",
          ],
          "links": Array [
            Object {
              "codename": "main_project",
              "linkId": "1a3aa6bd-7b4b-486f-86ab-7bfe359ad614",
              "type": "project",
              "urlSlug": "",
            },
          ],
          "name": "Content",
          "resolvedData": Object {
            "componentCodenames": Array [
              "n26ddfb46_c2de_010e_3ad9_ec9aa8955144",
            ],
            "html": "<h1>Landing page</h1>
<p><br>
<strong>Lorem </strong><em>ipsum </em><sup>dolor </sup><sub>sit </sub>amet, consectetuer adipiscing elit. Duis condimentum augue id magna semper rutrum. Phasellus faucibus molestie nisl.&nbsp;</p>
<ol>
  <li>Aliquam erat volutpat.&nbsp;</li>
</ol>
<p><br></p>
<p><a href=\\"https://kontent.ai\\">Link to web.</a></p>
<p><br></p>
<p><a data-asset-id=\\"98bf37b8-8be3-45e9-b284-293e205bf9a3\\" href=\\"https://assets-eu-01.kc-usercontent.com:443/71be1cca-0be3-0159-fdfe-6cf11b092e68/6c7c8367-f38c-41a1-a41d-91a71dcc073a/Back%203_4%20right.jpg\\">Link to Image</a></p>
<p><br></p>
<figure data-asset-id=\\"98bf37b8-8be3-45e9-b284-293e205bf9a3\\" data-image-id=\\"98bf37b8-8be3-45e9-b284-293e205bf9a3\\"><img src=\\"https://assets-eu-01.kc-usercontent.com:443/71be1cca-0be3-0159-fdfe-6cf11b092e68/6c7c8367-f38c-41a1-a41d-91a71dcc073a/Back%203_4%20right.jpg\\" data-asset-id=\\"98bf37b8-8be3-45e9-b284-293e205bf9a3\\" data-image-id=\\"98bf37b8-8be3-45e9-b284-293e205bf9a3\\" alt=\\"\\"></figure>
<p>Inline component</p>
<p type=\\"application/kenticocloud\\" data-type=\\"item\\" data-rel=\\"component\\" data-codename=\\"n26ddfb46_c2de_010e_3ad9_ec9aa8955144\\" data-sdk-resolved=\\"1\\" class=\\"kc-linked-item-wrapper\\">###landing_page_image_section###</p>
<p><br></p>
<p>Projects (Link to content item):</p>
<ul>
  <li><a data-item-id=\\"1a3aa6bd-7b4b-486f-86ab-7bfe359ad614\\" href=\\"###projectlink###\\">Main project</a></li>
</ul>
<p>Aliquam ante. Nam quis nulla. Sed ac dolor sit amet purus malesuada congue. Curabitur vitae diam non enim vestibulum interdum. Phasellus et lorem id felis nonummy placerat. Etiam posuere lacus quis dolor.&nbsp;</p>
<p><strong>Project (Link to existing content item)</strong></p>
<p type=\\"application/kenticocloud\\" data-type=\\"item\\" data-rel=\\"link\\" data-codename=\\"main_project\\" data-sdk-resolved=\\"1\\" class=\\"kc-linked-item-wrapper\\">###project###</p>",
            "linkedItemCodenames": Array [
              "main_project",
            ],
          },
          "type": "rich_text",
          "value": "<h1>Landing page</h1>
<p><br>
<strong>Lorem </strong><em>ipsum </em><sup>dolor </sup><sub>sit </sub>amet, consectetuer adipiscing elit. Duis condimentum augue id magna semper rutrum. Phasellus faucibus molestie nisl.&nbsp;</p>
<ol>
  <li>Aliquam erat volutpat.&nbsp;</li>
</ol>
<p><br></p>
<p><a href=\\"https://kontent.ai\\">Link to web.</a></p>
<p><br></p>
<p><a data-asset-id=\\"98bf37b8-8be3-45e9-b284-293e205bf9a3\\" href=\\"https://assets-eu-01.kc-usercontent.com:443/71be1cca-0be3-0159-fdfe-6cf11b092e68/6c7c8367-f38c-41a1-a41d-91a71dcc073a/Back%203_4%20right.jpg\\">Link to Image</a></p>
<p><br></p>
<figure data-asset-id=\\"98bf37b8-8be3-45e9-b284-293e205bf9a3\\" data-image-id=\\"98bf37b8-8be3-45e9-b284-293e205bf9a3\\"><img src=\\"https://assets-eu-01.kc-usercontent.com:443/71be1cca-0be3-0159-fdfe-6cf11b092e68/6c7c8367-f38c-41a1-a41d-91a71dcc073a/Back%203_4%20right.jpg\\" data-asset-id=\\"98bf37b8-8be3-45e9-b284-293e205bf9a3\\" data-image-id=\\"98bf37b8-8be3-45e9-b284-293e205bf9a3\\" alt=\\"\\"></figure>
<p>Inline component</p>
<object type=\\"application/kenticocloud\\" data-type=\\"item\\" data-rel=\\"component\\" data-codename=\\"n26ddfb46_c2de_010e_3ad9_ec9aa8955144\\"></object>
<p><br></p>
<p>Projects (Link to content item):</p>
<ul>
  <li><a data-item-id=\\"1a3aa6bd-7b4b-486f-86ab-7bfe359ad614\\" href=\\"\\">Main project</a></li>
</ul>
<p>Aliquam ante. Nam quis nulla. Sed ac dolor sit amet purus malesuada congue. Curabitur vitae diam non enim vestibulum interdum. Phasellus et lorem id felis nonummy placerat. Etiam posuere lacus quis dolor.&nbsp;</p>
<p><strong>Project (Link to existing content item)</strong></p>
<object type=\\"application/kenticocloud\\" data-type=\\"item\\" data-rel=\\"link\\" data-codename=\\"main_project\\"></object>",
        },
      },
      "id": "dummy-kentico-kontent-item-simple-landing-page-default",
      "internal": Object {
        "contentDigest": "16d6806589cc9c3f25e4bd43d9fff1ab",
        "type": "KontentItemLandingPage",
      },
      "otherLanguages___NODE": Array [],
      "parent": null,
      "preferred_language": "default",
      "system": Object {
        "codename": "simple_landing_page",
        "id": "45267fb2-2e53-41de-a51f-77a28f0a05e5",
        "language": "default",
        "lastModified": "2019-10-09T11:29:41.355Z",
        "name": "Simple Landing page",
        "sitemapLocations": Array [],
        "type": "landing_page",
      },
      "usedByContentItems___NODE": Array [],
    },
  ],
  Array [
    Object {
      "children": Array [],
      "contentType___NODE": "dummy-kentico-kontent-type-landing-page-image-section",
      "elements": Object {
        "images": Object {
          "name": "Images",
          "type": "asset",
          "value": Array [
            Object {
              "contract": Object {
                "description": null,
                "height": 1700,
                "name": "Back 3_4 right.jpg",
                "size": 1639188,
                "type": "image/jpeg",
                "url": "https://assets-eu-01.kc-usercontent.com:443/71be1cca-0be3-0159-fdfe-6cf11b092e68/6c7c8367-f38c-41a1-a41d-91a71dcc073a/Back%203_4%20right.jpg",
                "width": 3010,
              },
              "description": null,
              "height": 1700,
              "name": "Back 3_4 right.jpg",
              "size": 1639188,
              "type": "image/jpeg",
              "url": "https://assets-eu-01.kc-usercontent.com:443/71be1cca-0be3-0159-fdfe-6cf11b092e68/6c7c8367-f38c-41a1-a41d-91a71dcc073a/Back%203_4%20right.jpg",
              "width": 3010,
            },
            Object {
              "contract": Object {
                "description": null,
                "height": 1826,
                "name": "Front.jpg",
                "size": 1552869,
                "type": "image/jpeg",
                "url": "https://assets-eu-01.kc-usercontent.com:443/71be1cca-0be3-0159-fdfe-6cf11b092e68/1b6e8055-2952-4236-92d3-33e7774fb7d0/Front.jpg",
                "width": 2365,
              },
              "description": null,
              "height": 1826,
              "name": "Front.jpg",
              "size": 1552869,
              "type": "image/jpeg",
              "url": "https://assets-eu-01.kc-usercontent.com:443/71be1cca-0be3-0159-fdfe-6cf11b092e68/1b6e8055-2952-4236-92d3-33e7774fb7d0/Front.jpg",
              "width": 2365,
            },
            Object {
              "contract": Object {
                "description": null,
                "height": 1700,
                "name": "Back 3_4.jpg",
                "size": 1653072,
                "type": "image/jpeg",
                "url": "https://assets-eu-01.kc-usercontent.com:443/71be1cca-0be3-0159-fdfe-6cf11b092e68/feb9cb93-0d5e-4a22-879f-f6cb7dd6b67b/Back%203_4.jpg",
                "width": 3010,
              },
              "description": null,
              "height": 1700,
              "name": "Back 3_4.jpg",
              "size": 1653072,
              "type": "image/jpeg",
              "url": "https://assets-eu-01.kc-usercontent.com:443/71be1cca-0be3-0159-fdfe-6cf11b092e68/feb9cb93-0d5e-4a22-879f-f6cb7dd6b67b/Back%203_4.jpg",
              "width": 3010,
            },
          ],
        },
        "title": Object {
          "name": "Title",
          "type": "text",
          "value": "Cap",
        },
      },
      "id": "dummy-kentico-kontent-item-n26ddfb46-c2de-010e-3ad9-ec9aa8955144-default",
      "internal": Object {
        "contentDigest": "09ec3b54b529f0fab8deac89b2973886",
        "type": "KontentItemLandingPageImageSection",
      },
      "otherLanguages___NODE": Array [],
      "parent": null,
      "preferred_language": "default",
      "system": Object {
        "codename": "n26ddfb46_c2de_010e_3ad9_ec9aa8955144",
        "id": "26ddfb46-c2de-010e-3ad9-ec9aa8955144",
        "language": "default",
        "lastModified": "2019-10-09T11:29:41.355Z",
        "name": "26ddfb46-c2de-010e-3ad9-ec9aa8955144",
        "sitemapLocations": Array [],
        "type": "landing_page_image_section",
      },
      "usedByContentItems___NODE": Array [
        "dummy-kentico-kontent-item-simple-landing-page-default",
      ],
    },
  ],
]
`;
